From 2042d411c5eb6f1bb117744225efe7b35e2a17e3 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 9 Jul 2025 14:37:16 +0200
Subject: [PATCH] arm64: dts: st: introduce the debug bus for stm32mp2x
 platforms

Some peripherals cannot be probed if a debug configuration is not set
in the BSEC.
Introduce a debug bus that will check the debug subsystem accessibility
before probing these peripheral drivers. Consequently, move all
peripheral relying only on the debug domain access under the debug bus.
Also list the access controllers of the HDP node.

Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I48f1eebfe56827e8f6d0a6672c691a0d8405c8a6
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/473655
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 518 +++++++++++++------------
 arch/arm64/boot/dts/st/stm32mp231.dtsi | 430 ++++++++++----------
 arch/arm64/boot/dts/st/stm32mp233.dtsi |  80 ++--
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 430 ++++++++++----------
 arch/arm64/boot/dts/st/stm32mp253.dtsi |  80 ++--
 5 files changed, 807 insertions(+), 731 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -80,38 +80,6 @@
 		};
 	};
 
-	cs_replicator: replicator {
-		compatible = "arm,coresight-static-replicator";
-		status = "disabled";
-
-		in-ports {
-			port {
-				replicator_in_port: endpoint {
-					remote-endpoint = <&etf_out_port>;
-				};
-			};
-		};
-
-		out-ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			port@0 {
-				reg = <0>;
-				replicator_out_port0: endpoint {
-					remote-endpoint = <&etr_in_port>;
-				};
-			};
-
-			port@1 {
-				reg = <1>;
-				replicator_out_port1: endpoint {
-					remote-endpoint = <&tpiu_in_port>;
-				};
-			};
-		};
-	};
-
 	firmware {
 		optee: optee {
 			bootph-all;
@@ -2016,6 +1984,7 @@
 			compatible = "st,stm32mp-hdp";
 			reg = <0x44090000 0x0 0x400>;
 			clocks = <&rcc CK_BUS_HDP>;
+			access-controllers = <&dbg_bus 1>, <&rifsc 57>;
 			status = "disabled";
 		};
 
@@ -2513,212 +2482,6 @@
 			reg = <0x48802000 0x0 0xac>;
 			status = "disabled";
 		};
-
-		cs_funnel: funnel@4a010000 {
-			compatible = "arm,coresight-dynamic-funnel", "arm,primecell";
-			reg = <0x4a010000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk";
-			status = "disabled";
-
-			in-ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				port@0 {
-					reg = <0>;
-					funnel_in_port0: endpoint {
-						remote-endpoint = <&etm0_out_port>;
-					};
-				};
-
-				port@2 {
-					reg = <2>;
-					funnel_in_port2: endpoint {
-						remote-endpoint = <&stm_out_port>;
-					};
-				};
-			};
-
-			out-ports {
-				port {
-					funnel_out_port: endpoint {
-						remote-endpoint = <&etf_in_port>;
-					};
-				};
-			};
-		};
-
-		cs_etf: etf@4a020000 {
-			compatible = "arm,coresight-tmc", "arm,primecell";
-			reg = <0x4a020000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk";
-			status = "disabled";
-
-			in-ports {
-				port {
-					etf_in_port: endpoint {
-						remote-endpoint = <&funnel_out_port>;
-					};
-				};
-			};
-
-			out-ports {
-				port {
-					etf_out_port: endpoint {
-						remote-endpoint = <&replicator_in_port>;
-					};
-				};
-			};
-		};
-
-		cs_tpiu: tpiu@4a040000 {
-			compatible = "arm,coresight-tpiu", "arm,primecell";
-			reg = <0x4a040000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_TPIU>;
-			clock-names = "apb_pclk", "atclk";
-			status = "disabled";
-
-			in-ports {
-				port {
-					tpiu_in_port: endpoint {
-						remote-endpoint = <&replicator_out_port1>;
-					};
-				};
-			};
-		};
-
-		cs_stm: stm@4a070000 {
-			compatible = "arm,coresight-stm", "arm,primecell";
-			reg = <0x4a070000 0x0 0x1000>,
-			      <0x4a800000 0x0 0x400000>;
-			reg-names = "stm-base", "stm-stimulus-base";
-			clocks = <&scmi_clk CK_SCMI_BUS_STM>, <&scmi_clk CK_SCMI_KER_STM>;
-			clock-names = "apb_pclk", "atclk";
-			status = "disabled";
-
-			out-ports {
-				port {
-					stm_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port2>;
-					};
-				};
-			};
-		};
-
-		cs_cti0: cti@4a080000 {
-			compatible = "arm,coresight-cti", "arm,primecell";
-			reg = <0x4a080000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
-
-			trig-conns@0 {
-				reg = <0>;
-				arm,trig-in-sigs = <0 1>;
-				arm,trig-in-types = <SNK_FULL
-						     SNK_ACQCOMP>;
-				arm,trig-out-sigs = <0 1>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_etr>;
-			};
-
-			trig-conns@1 {
-				reg = <1>;
-				arm,trig-in-sigs = <2 3>;
-				arm,trig-in-types = <SNK_FULL
-						     SNK_ACQCOMP>;
-				arm,trig-out-sigs = <2 3>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_etf>;
-			};
-
-			trig-conns@2 {
-				reg = <2>;
-				arm,trig-out-sigs = <4 5>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_tpiu>;
-			};
-
-			trig-conns@3 {
-				reg = <3>;
-				arm,trig-in-sigs = <4 5 6 7>;
-				arm,trig-in-types = <STM_TOUT_SPTE
-						     STM_TOUT_SW
-						     STM_TOUT_HETE
-						     STM_ASYNCOUT>;
-				arm,cs-dev-assoc = <&cs_stm>;
-			};
-		};
-
-		cs_cti1: cti@4a090000 {
-			compatible = "arm,coresight-cti", "arm,primecell";
-			reg = <0x4a090000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
-
-			trig-conns@0 {
-				reg = <0>;
-				arm,trig-in-sigs = <0>;
-				arm,trig-in-types = <GEN_IO>;
-				arm,trig-out-sigs = <0>;
-				arm,trig-out-types = <GEN_IO>;
-				arm,trig-conn-name = "dbtrgio";
-			};
-
-			trig-conns@1 {
-				reg = <1>;
-				arm,trig-out-sigs = <1 2>;
-				arm,trig-out-types = <STM_HWEVENT
-						      STM_HWEVENT>;
-				arm,cs-dev-assoc = <&cs_stm>;
-			};
-		};
-
-		cs_cpu_debug0: cpu-debug@4a210000 {
-			compatible = "arm,coresight-cpu-debug", "arm,primecell";
-			reg = <0x4a210000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			status = "disabled";
-		};
-
-		cs_cti_cpu0: cti@4a220000 {
-			compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
-				     "arm,primecell";
-			reg = <0x4a220000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			cpu = <&cpu0>;
-			arm,cs-dev-assoc = <&cs_etm0>;
-			status = "disabled";
-		};
-
-		cs_etm0: etm@4a240000 {
-			compatible = "arm,coresight-etm4x", "arm,primecell";
-			reg = <0x4a240000 0x0 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk", "atclk";
-			cpu = <&cpu0>;
-			status = "disabled";
-
-			out-ports {
-				port {
-					etm0_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port0>;
-					};
-				};
-			};
-		};
 	};
 
 	soc1: soc@1 {
@@ -2795,17 +2558,278 @@
 			};
 		};
 
-		cs_etr: etr@4a030000 {
-			compatible = "arm,coresight-tmc", "arm,primecell";
-			reg = <0x4a030000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_BUS_ETR>, <&scmi_clk CK_SCMI_KER_ETR>;
-			clock-names = "apb_pclk", "atclk";
+		dbg_bus: bus@4a000000 {
+			compatible = "st,stm32mp211-dbg-bus";
+			reg = <0x4a000000 0x800000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			clocks = <&scmi_clk CK_SCMI_SYSATB>;
+			ranges;
+			#access-controller-cells = <1>;
 			status = "disabled";
 
-			in-ports {
-				port {
-					etr_in_port: endpoint {
-						remote-endpoint = <&replicator_out_port0>;
+			cs_funnel: funnel@4a010000 {
+				compatible = "arm,coresight-dynamic-funnel", "arm,primecell";
+				reg = <0x4a010000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						funnel_in_port0: endpoint {
+							remote-endpoint = <&etm0_out_port>;
+						};
+					};
+
+					port@2 {
+						reg = <2>;
+						funnel_in_port2: endpoint {
+							remote-endpoint = <&stm_out_port>;
+						};
+					};
+				};
+
+				out-ports {
+					port {
+						funnel_out_port: endpoint {
+							remote-endpoint = <&etf_in_port>;
+						};
+					};
+				};
+			};
+
+			cs_etf: etf@4a020000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x4a020000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						etf_in_port: endpoint {
+							remote-endpoint = <&funnel_out_port>;
+						};
+					};
+				};
+
+				out-ports {
+					port {
+						etf_out_port: endpoint {
+							remote-endpoint = <&replicator_in_port>;
+						};
+					};
+				};
+			};
+
+			cs_etr: etr@4a030000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x4a030000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_BUS_ETR>, <&scmi_clk CK_SCMI_KER_ETR>;
+				clock-names = "apb_pclk", "atclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						etr_in_port: endpoint {
+							remote-endpoint = <&replicator_out_port0>;
+						};
+					};
+				};
+			};
+
+			cs_tpiu: tpiu@4a040000 {
+				compatible = "arm,coresight-tpiu", "arm,primecell";
+				reg = <0x4a040000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_TPIU>;
+				clock-names = "apb_pclk", "atclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						tpiu_in_port: endpoint {
+							remote-endpoint = <&replicator_out_port1>;
+						};
+					};
+				};
+			};
+
+			cs_replicator: replicator@4a050000 {
+				compatible = "arm,coresight-static-replicator";
+				access-controllers = <&dbg_bus 0>;
+				reg = <0x4a050000 0x10000>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						replicator_in_port: endpoint {
+							remote-endpoint = <&etf_out_port>;
+						};
+					};
+				};
+
+				out-ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						replicator_out_port0: endpoint {
+							remote-endpoint = <&etr_in_port>;
+						};
+					};
+
+					port@1 {
+						reg = <1>;
+						replicator_out_port1: endpoint {
+							remote-endpoint = <&tpiu_in_port>;
+						};
+					};
+				};
+			};
+
+			cs_stm: stm@4a070000 {
+				compatible = "arm,coresight-stm", "arm,primecell";
+				reg = <0x4a070000 0x1000>,
+				<0x4a800000 0x400000>;
+				reg-names = "stm-base", "stm-stimulus-base";
+				clocks = <&scmi_clk CK_SCMI_BUS_STM>, <&scmi_clk CK_SCMI_KER_STM>;
+				clock-names = "apb_pclk", "atclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				out-ports {
+					port {
+						stm_out_port: endpoint {
+							remote-endpoint = <&funnel_in_port2>;
+						};
+					};
+				};
+			};
+
+			cs_cti0: cti@4a080000 {
+				compatible = "arm,coresight-cti", "arm,primecell";
+				reg = <0x4a080000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				trig-conns@0 {
+					reg = <0>;
+					arm,trig-in-sigs = <0 1>;
+					arm,trig-in-types = <SNK_FULL
+							     SNK_ACQCOMP>;
+					arm,trig-out-sigs = <0 1>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_etr>;
+				};
+
+				trig-conns@1 {
+					reg = <1>;
+					arm,trig-in-sigs = <2 3>;
+					arm,trig-in-types = <SNK_FULL
+							     SNK_ACQCOMP>;
+					arm,trig-out-sigs = <2 3>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_etf>;
+				};
+
+				trig-conns@2 {
+					reg = <2>;
+					arm,trig-out-sigs = <4 5>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_tpiu>;
+				};
+
+				trig-conns@3 {
+					reg = <3>;
+					arm,trig-in-sigs = <4 5 6 7>;
+					arm,trig-in-types = <STM_TOUT_SPTE
+							     STM_TOUT_SW
+							     STM_TOUT_HETE
+							     STM_ASYNCOUT>;
+					arm,cs-dev-assoc = <&cs_stm>;
+				};
+			};
+
+			cs_cti1: cti@4a090000 {
+				compatible = "arm,coresight-cti", "arm,primecell";
+				reg = <0x4a090000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				trig-conns@0 {
+					reg = <0>;
+					arm,trig-in-sigs = <0>;
+					arm,trig-in-types = <GEN_IO>;
+					arm,trig-out-sigs = <0>;
+					arm,trig-out-types = <GEN_IO>;
+					arm,trig-conn-name = "dbtrgio";
+				};
+
+				trig-conns@1 {
+					reg = <1>;
+					arm,trig-out-sigs = <1 2>;
+					arm,trig-out-types = <STM_HWEVENT
+							      STM_HWEVENT>;
+					arm,cs-dev-assoc = <&cs_stm>;
+				};
+			};
+
+			cs_cpu_debug0: cpu-debug@4a210000 {
+				compatible = "arm,coresight-cpu-debug", "arm,primecell";
+				reg = <0x4a210000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+			};
+
+			cs_cti_cpu0: cti@4a220000 {
+				compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
+					"arm,primecell";
+				reg = <0x4a220000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				cpu = <&cpu0>;
+				arm,cs-dev-assoc = <&cs_etm0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+			};
+
+			cs_etm0: etm@4a240000 {
+				compatible = "arm,coresight-etm4x", "arm,primecell";
+				reg = <0x4a240000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk", "atclk";
+				cpu = <&cpu0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				out-ports {
+					port {
+						etm0_out_port: endpoint {
+							remote-endpoint = <&funnel_in_port0>;
+						};
 					};
 				};
 			};
--- a/arch/arm64/boot/dts/st/stm32mp231.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp231.dtsi
@@ -89,38 +89,6 @@
 		};
 	};
 
-	cs_replicator: replicator {
-		compatible = "arm,coresight-static-replicator";
-		status = "disabled";
-
-		in-ports {
-			port {
-				replicator_in_port: endpoint {
-					remote-endpoint = <&etf_out_port>;
-				};
-			};
-		};
-
-		out-ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			port@0 {
-				reg = <0>;
-				replicator_out_port0: endpoint {
-					remote-endpoint = <&etr_in_port>;
-				};
-			};
-
-			port@1 {
-				reg = <1>;
-				replicator_out_port1: endpoint {
-					remote-endpoint = <&tpiu_in_port>;
-				};
-			};
-		};
-	};
-
 	firmware {
 		optee: optee {
 			bootph-all;
@@ -2246,6 +2214,7 @@
 			compatible = "st,stm32mp-hdp";
 			reg = <0x44090000 0x400>;
 			clocks = <&rcc CK_BUS_HDP>;
+			access-controllers = <&dbg_bus 1>, <&rifsc 57>;
 			status = "disabled";
 		};
 
@@ -2789,225 +2758,280 @@
 			status = "disabled";
 		};
 
-		cs_funnel: funnel@4a020000 {
-			compatible = "arm,coresight-dynamic-funnel", "arm,primecell";
-			reg = <0x4a020000 0x1000>;
+		dbg_bus: bus@4a000000 {
+			compatible = "st,stm32mp231-dbg-bus";
+			reg = <0x4a000000 0x800000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
 			clocks = <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk";
+			ranges;
+			#access-controller-cells = <1>;
 			status = "disabled";
 
-			in-ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
+			cs_funnel: funnel@4a020000 {
+				compatible = "arm,coresight-dynamic-funnel", "arm,primecell";
+				reg = <0x4a020000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-				port@0 {
-					reg = <0>;
-					funnel_in_port0: endpoint {
-						remote-endpoint = <&etm0_out_port>;
+				in-ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						funnel_in_port0: endpoint {
+							remote-endpoint = <&etm0_out_port>;
+						};
+					};
+
+					port@2 {
+						reg = <2>;
+						funnel_in_port2: endpoint {
+							remote-endpoint = <&stm_out_port>;
+						};
 					};
 				};
 
-				port@2 {
-					reg = <2>;
-					funnel_in_port2: endpoint {
-						remote-endpoint = <&stm_out_port>;
+				out-ports {
+					port {
+						funnel_out_port: endpoint {
+							remote-endpoint = <&etf_in_port>;
+						};
 					};
 				};
 			};
 
-			out-ports {
-				port {
-					funnel_out_port: endpoint {
-						remote-endpoint = <&etf_in_port>;
+			cs_etf: etf@4a030000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x4a030000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						etf_in_port: endpoint {
+							remote-endpoint = <&funnel_out_port>;
+						};
+					};
+				};
+
+				out-ports {
+					port {
+						etf_out_port: endpoint {
+							remote-endpoint = <&replicator_in_port>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_etf: etf@4a030000 {
-			compatible = "arm,coresight-tmc", "arm,primecell";
-			reg = <0x4a030000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk";
-			status = "disabled";
+			cs_etr: etr@4a040000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x4a040000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_BUS_ETR>, <&scmi_clk CK_SCMI_KER_ETR>;
+				clock-names = "apb_pclk", "atclk";
+				arm,max-burst-size = <3>;
+				arm,scatter-gather;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-			in-ports {
-				port {
-					etf_in_port: endpoint {
-						remote-endpoint = <&funnel_out_port>;
+				in-ports {
+					port {
+						etr_in_port: endpoint {
+							remote-endpoint = <&replicator_out_port0>;
+						};
 					};
 				};
 			};
 
-			out-ports {
-				port {
-					etf_out_port: endpoint {
-						remote-endpoint = <&replicator_in_port>;
+			cs_tpiu: tpiu@4a050000 {
+				compatible = "arm,coresight-tpiu", "arm,primecell";
+				reg = <0x4a050000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_TPIU>;
+				clock-names = "apb_pclk", "atclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						tpiu_in_port: endpoint {
+							remote-endpoint = <&replicator_out_port1>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_etr: etr@4a040000 {
-			compatible = "arm,coresight-tmc", "arm,primecell";
-			reg = <0x4a040000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_BUS_ETR>, <&scmi_clk CK_SCMI_KER_ETR>;
-			clock-names = "apb_pclk", "atclk";
-			arm,max-burst-size = <3>;
-			arm,scatter-gather;
-			status = "disabled";
+			cs_replicator: replicator@4a060000 {
+				compatible = "arm,coresight-static-replicator";
+				reg = <0x4a060000 0x10000>;
+				access-controllers = <&dbg_bus 0>;
+				status = "okay";
 
-			in-ports {
-				port {
-					etr_in_port: endpoint {
-						remote-endpoint = <&replicator_out_port0>;
+				in-ports {
+					port {
+						replicator_in_port: endpoint {
+							remote-endpoint = <&etf_out_port>;
+						};
 					};
 				};
-			};
-		};
 
-		cs_tpiu: tpiu@4a050000 {
-			compatible = "arm,coresight-tpiu", "arm,primecell";
-			reg = <0x4a050000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_TPIU>;
-			clock-names = "apb_pclk", "atclk";
-			status = "disabled";
+				out-ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
 
-			in-ports {
-				port {
-					tpiu_in_port: endpoint {
-						remote-endpoint = <&replicator_out_port1>;
+					port@0 {
+						reg = <0>;
+						replicator_out_port0: endpoint {
+							remote-endpoint = <&etr_in_port>;
+						};
+					};
+
+					port@1 {
+						reg = <1>;
+						replicator_out_port1: endpoint {
+							remote-endpoint = <&tpiu_in_port>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_stm: stm@4a080000 {
-			compatible = "arm,coresight-stm", "arm,primecell";
-			reg = <0x4a080000 0x1000>,
-			      <0x4a800000 0x400000>;
-			reg-names = "stm-base", "stm-stimulus-base";
-			clocks = <&scmi_clk CK_SCMI_BUS_STM>, <&scmi_clk CK_SCMI_KER_STM>;
-			clock-names = "apb_pclk", "atclk";
-			status = "disabled";
+			cs_stm: stm@4a080000 {
+				compatible = "arm,coresight-stm", "arm,primecell";
+				reg = <0x4a080000 0x1000>,
+				      <0x4a800000 0x400000>;
+				reg-names = "stm-base", "stm-stimulus-base";
+				clocks = <&scmi_clk CK_SCMI_BUS_STM>, <&scmi_clk CK_SCMI_KER_STM>;
+				clock-names = "apb_pclk", "atclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-			out-ports {
-				port {
-					stm_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port2>;
+				out-ports {
+					port {
+						stm_out_port: endpoint {
+							remote-endpoint = <&funnel_in_port2>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_cti0: cti@4a090000 {
-			compatible = "arm,coresight-cti", "arm,primecell";
-			reg = <0x4a090000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
+			cs_cti0: cti@4a090000 {
+				compatible = "arm,coresight-cti", "arm,primecell";
+				reg = <0x4a090000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-			trig-conns@0 {
-				reg = <0>;
-				arm,trig-in-sigs = <0 1>;
-				arm,trig-in-types = <SNK_FULL
-						     SNK_ACQCOMP>;
-				arm,trig-out-sigs = <0 1>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_etr>;
-			};
-
-			trig-conns@1 {
-				reg = <1>;
-				arm,trig-in-sigs = <2 3>;
-				arm,trig-in-types = <SNK_FULL
-						     SNK_ACQCOMP>;
-				arm,trig-out-sigs = <2 3>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_etf>;
-			};
-
-			trig-conns@2 {
-				reg = <2>;
-				arm,trig-out-sigs = <4 5>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_tpiu>;
-			};
-
-			trig-conns@3 {
-				reg = <3>;
-				arm,trig-in-sigs = <4 5 6 7>;
-				arm,trig-in-types = <STM_TOUT_SPTE
-						     STM_TOUT_SW
-						     STM_TOUT_HETE
-						     STM_ASYNCOUT>;
-				arm,cs-dev-assoc = <&cs_stm>;
-			};
-		};
+				trig-conns@0 {
+					reg = <0>;
+					arm,trig-in-sigs = <0 1>;
+					arm,trig-in-types = <SNK_FULL
+							     SNK_ACQCOMP>;
+					arm,trig-out-sigs = <0 1>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_etr>;
+				};
 
-		cs_cti1: cti@4a0a0000 {
-			compatible = "arm,coresight-cti", "arm,primecell";
-			reg = <0x4a0a0000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
+				trig-conns@1 {
+					reg = <1>;
+					arm,trig-in-sigs = <2 3>;
+					arm,trig-in-types = <SNK_FULL
+							     SNK_ACQCOMP>;
+					arm,trig-out-sigs = <2 3>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_etf>;
+				};
+
+				trig-conns@2 {
+					reg = <2>;
+					arm,trig-out-sigs = <4 5>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_tpiu>;
+				};
 
-			trig-conns@0 {
-				reg = <0>;
-				arm,trig-in-sigs = <0>;
-				arm,trig-in-types = <GEN_IO>;
-				arm,trig-out-sigs = <0>;
-				arm,trig-out-types = <GEN_IO>;
-				arm,trig-conn-name = "dbtrgio";
-			};
-
-			trig-conns@1 {
-				reg = <1>;
-				arm,trig-out-sigs = <1 2>;
-				arm,trig-out-types = <STM_HWEVENT
-						      STM_HWEVENT>;
-				arm,cs-dev-assoc = <&cs_stm>;
+				trig-conns@3 {
+					reg = <3>;
+					arm,trig-in-sigs = <4 5 6 7>;
+					arm,trig-in-types = <STM_TOUT_SPTE
+							     STM_TOUT_SW
+							     STM_TOUT_HETE
+							     STM_ASYNCOUT>;
+					arm,cs-dev-assoc = <&cs_stm>;
+				};
 			};
-		};
 
-		cs_cpu_debug0: cpu-debug@4a210000 {
-			compatible = "arm,coresight-cpu-debug", "arm,primecell";
-			reg = <0x4a210000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			status = "disabled";
-		};
+			cs_cti1: cti@4a0a0000 {
+				compatible = "arm,coresight-cti", "arm,primecell";
+				reg = <0x4a0a0000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-		cs_cti_cpu0: cti@4a220000 {
-			compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
-				     "arm,primecell";
-			reg = <0x4a220000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			cpu = <&cpu0>;
-			arm,cs-dev-assoc = <&cs_etm0>;
-			status = "disabled";
-		};
+				trig-conns@0 {
+					reg = <0>;
+					arm,trig-in-sigs = <0>;
+					arm,trig-in-types = <GEN_IO>;
+					arm,trig-out-sigs = <0>;
+					arm,trig-out-types = <GEN_IO>;
+					arm,trig-conn-name = "dbtrgio";
+				};
 
-		cs_etm0: etm@4a240000 {
-			compatible = "arm,coresight-etm4x", "arm,primecell";
-			reg = <0x4a240000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk", "atclk";
-			cpu = <&cpu0>;
-			status = "disabled";
+				trig-conns@1 {
+					reg = <1>;
+					arm,trig-out-sigs = <1 2>;
+					arm,trig-out-types = <STM_HWEVENT
+							      STM_HWEVENT>;
+					arm,cs-dev-assoc = <&cs_stm>;
+				};
+			};
+
+			cs_cpu_debug0: cpu-debug@4a210000 {
+				compatible = "arm,coresight-cpu-debug", "arm,primecell";
+				reg = <0x4a210000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+			};
+
+			cs_cti_cpu0: cti@4a220000 {
+				compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
+					     "arm,primecell";
+				reg = <0x4a220000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				cpu = <&cpu0>;
+				arm,cs-dev-assoc = <&cs_etm0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+			};
 
-			out-ports {
-				port {
-					etm0_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port0>;
+			cs_etm0: etm@4a240000 {
+				compatible = "arm,coresight-etm4x", "arm,primecell";
+				reg = <0x4a240000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk", "atclk";
+				cpu = <&cpu0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				out-ports {
+					port {
+						etm0_out_port: endpoint {
+							remote-endpoint = <&funnel_in_port0>;
+						};
 					};
 				};
 			};
--- a/arch/arm64/boot/dts/st/stm32mp233.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp233.dtsi
@@ -37,45 +37,6 @@
 			     <GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
 	};
-
-	soc@0 {
-		cs_cpu_debug1: cpu-debug@4a310000 {
-			compatible = "arm,coresight-cpu-debug", "arm,primecell";
-			reg = <0x4a310000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			cpu = <&cpu1>;
-			status = "disabled";
-		};
-
-		cs_cti_cpu1: cti@4a320000 {
-			compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
-				     "arm,primecell";
-			reg = <0x4a320000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			cpu = <&cpu1>;
-			arm,cs-dev-assoc = <&cs_etm1>;
-			status = "disabled";
-		};
-
-		cs_etm1: etm@4a340000 {
-			compatible = "arm,coresight-etm4x", "arm,primecell";
-			reg = <0x4a340000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk", "atclk";
-			cpu = <&cpu1>;
-			status = "disabled";
-
-			out-ports {
-				port {
-					etm1_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port1>;
-					};
-				};
-			};
-		};
-	};
 };
 
 &cs_funnel {
@@ -87,6 +48,47 @@
 			};
 		};
 	};
+};
+
+&dbg_bus {
+	cs_cpu_debug1: cpu-debug@4a310000 {
+		compatible = "arm,coresight-cpu-debug", "arm,primecell";
+		reg = <0x4a310000 0x1000>;
+		clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+		clock-names = "apb_pclk";
+		cpu = <&cpu1>;
+		access-controllers = <&dbg_bus 0>;
+		status = "disabled";
+	};
+
+	cs_cti_cpu1: cti@4a320000 {
+		compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti", "arm,primecell";
+		reg = <0x4a320000 0x1000>;
+		clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+		clock-names = "apb_pclk";
+		cpu = <&cpu1>;
+		arm,cs-dev-assoc = <&cs_etm1>;
+		access-controllers = <&dbg_bus 0>;
+		status = "disabled";
+	};
+
+	cs_etm1: etm@4a340000 {
+		compatible = "arm,coresight-etm4x", "arm,primecell";
+		reg = <0x4a340000 0x1000>;
+		clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
+		clock-names = "apb_pclk", "atclk";
+		cpu = <&cpu1>;
+		access-controllers = <&dbg_bus 0>;
+		status = "disabled";
+
+		out-ports {
+			port {
+				etm1_out_port: endpoint {
+					remote-endpoint = <&funnel_in_port1>;
+				};
+			};
+		};
+	};
 };
 
 &intc {
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -89,38 +89,6 @@
 		};
 	};
 
-	cs_replicator: replicator {
-		compatible = "arm,coresight-static-replicator";
-		status = "disabled";
-
-		in-ports {
-			port {
-				replicator_in_port: endpoint {
-					remote-endpoint = <&etf_out_port>;
-				};
-			};
-		};
-
-		out-ports {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			port@0 {
-				reg = <0>;
-				replicator_out_port0: endpoint {
-					remote-endpoint = <&etr_in_port>;
-				};
-			};
-
-			port@1 {
-				reg = <1>;
-				replicator_out_port1: endpoint {
-					remote-endpoint = <&tpiu_in_port>;
-				};
-			};
-		};
-	};
-
 	firmware {
 		optee: optee {
 			bootph-all;
@@ -2571,6 +2539,7 @@
 			compatible = "st,stm32mp-hdp";
 			reg = <0x44090000 0x400>;
 			clocks = <&rcc CK_BUS_HDP>;
+			access-controllers = <&dbg_bus 1>, <&rifsc 57>;
 			status = "disabled";
 		};
 
@@ -3151,225 +3120,280 @@
 			status = "disabled";
 		};
 
-		cs_funnel: funnel@4a020000 {
-			compatible = "arm,coresight-dynamic-funnel", "arm,primecell";
-			reg = <0x4a020000 0x1000>;
+		dbg_bus: bus@4a000000 {
+			compatible = "st,stm32mp251-dbg-bus";
+			reg = <0x4a000000 0x800000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
 			clocks = <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk";
+			ranges;
+			#access-controller-cells = <1>;
 			status = "disabled";
 
-			in-ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
+			cs_funnel: funnel@4a020000 {
+				compatible = "arm,coresight-dynamic-funnel", "arm,primecell";
+				reg = <0x4a020000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-				port@0 {
-					reg = <0>;
-					funnel_in_port0: endpoint {
-						remote-endpoint = <&etm0_out_port>;
+				in-ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					port@0 {
+						reg = <0>;
+						funnel_in_port0: endpoint {
+							remote-endpoint = <&etm0_out_port>;
+						};
+					};
+
+					port@2 {
+						reg = <2>;
+						funnel_in_port2: endpoint {
+							remote-endpoint = <&stm_out_port>;
+						};
 					};
 				};
 
-				port@2 {
-					reg = <2>;
-					funnel_in_port2: endpoint {
-						remote-endpoint = <&stm_out_port>;
+				out-ports {
+					port {
+						funnel_out_port: endpoint {
+							remote-endpoint = <&etf_in_port>;
+						};
 					};
 				};
 			};
 
-			out-ports {
-				port {
-					funnel_out_port: endpoint {
-						remote-endpoint = <&etf_in_port>;
+			cs_etf: etf@4a030000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x4a030000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						etf_in_port: endpoint {
+							remote-endpoint = <&funnel_out_port>;
+						};
+					};
+				};
+
+				out-ports {
+					port {
+						etf_out_port: endpoint {
+							remote-endpoint = <&replicator_in_port>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_etf: etf@4a030000 {
-			compatible = "arm,coresight-tmc", "arm,primecell";
-			reg = <0x4a030000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk";
-			status = "disabled";
+			cs_etr: etr@4a040000 {
+				compatible = "arm,coresight-tmc", "arm,primecell";
+				reg = <0x4a040000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_BUS_ETR>, <&scmi_clk CK_SCMI_KER_ETR>;
+				clock-names = "apb_pclk", "atclk";
+				arm,max-burst-size = <3>;
+				arm,scatter-gather;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-			in-ports {
-				port {
-					etf_in_port: endpoint {
-						remote-endpoint = <&funnel_out_port>;
+				in-ports {
+					port {
+						etr_in_port: endpoint {
+							remote-endpoint = <&replicator_out_port0>;
+						};
 					};
 				};
 			};
 
-			out-ports {
-				port {
-					etf_out_port: endpoint {
-						remote-endpoint = <&replicator_in_port>;
+			cs_tpiu: tpiu@4a050000 {
+				compatible = "arm,coresight-tpiu", "arm,primecell";
+				reg = <0x4a050000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_TPIU>;
+				clock-names = "apb_pclk", "atclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				in-ports {
+					port {
+						tpiu_in_port: endpoint {
+							remote-endpoint = <&replicator_out_port1>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_etr: etr@4a040000 {
-			compatible = "arm,coresight-tmc", "arm,primecell";
-			reg = <0x4a040000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_BUS_ETR>, <&scmi_clk CK_SCMI_KER_ETR>;
-			clock-names = "apb_pclk", "atclk";
-			arm,max-burst-size = <3>;
-			arm,scatter-gather;
-			status = "disabled";
+			cs_replicator: replicator@4a060000 {
+				compatible = "arm,coresight-static-replicator";
+				reg = <0x4a060000 0x10000>;
+				access-controllers = <&dbg_bus 0>;
+				status = "okay";
 
-			in-ports {
-				port {
-					etr_in_port: endpoint {
-						remote-endpoint = <&replicator_out_port0>;
+				in-ports {
+					port {
+						replicator_in_port: endpoint {
+							remote-endpoint = <&etf_out_port>;
+						};
 					};
 				};
-			};
-		};
 
-		cs_tpiu: tpiu@4a050000 {
-			compatible = "arm,coresight-tpiu", "arm,primecell";
-			reg = <0x4a050000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_TPIU>;
-			clock-names = "apb_pclk", "atclk";
-			status = "disabled";
+				out-ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
 
-			in-ports {
-				port {
-					tpiu_in_port: endpoint {
-						remote-endpoint = <&replicator_out_port1>;
+					port@0 {
+						reg = <0>;
+						replicator_out_port0: endpoint {
+							remote-endpoint = <&etr_in_port>;
+						};
+					};
+
+					port@1 {
+						reg = <1>;
+						replicator_out_port1: endpoint {
+							remote-endpoint = <&tpiu_in_port>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_stm: stm@4a080000 {
-			compatible = "arm,coresight-stm", "arm,primecell";
-			reg = <0x4a080000 0x1000>,
-			      <0x4a800000 0x400000>;
-			reg-names = "stm-base", "stm-stimulus-base";
-			clocks = <&scmi_clk CK_SCMI_BUS_STM>, <&scmi_clk CK_SCMI_KER_STM>;
-			clock-names = "apb_pclk", "atclk";
-			status = "disabled";
+			cs_stm: stm@4a080000 {
+				compatible = "arm,coresight-stm", "arm,primecell";
+				reg = <0x4a080000 0x1000>,
+				      <0x4a800000 0x400000>;
+				reg-names = "stm-base", "stm-stimulus-base";
+				clocks = <&scmi_clk CK_SCMI_BUS_STM>, <&scmi_clk CK_SCMI_KER_STM>;
+				clock-names = "apb_pclk", "atclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-			out-ports {
-				port {
-					stm_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port2>;
+				out-ports {
+					port {
+						stm_out_port: endpoint {
+							remote-endpoint = <&funnel_in_port2>;
+						};
 					};
 				};
 			};
-		};
 
-		cs_cti0: cti@4a090000 {
-			compatible = "arm,coresight-cti", "arm,primecell";
-			reg = <0x4a090000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
+			cs_cti0: cti@4a090000 {
+				compatible = "arm,coresight-cti", "arm,primecell";
+				reg = <0x4a090000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-			trig-conns@0 {
-				reg = <0>;
-				arm,trig-in-sigs = <0 1>;
-				arm,trig-in-types = <SNK_FULL
-						     SNK_ACQCOMP>;
-				arm,trig-out-sigs = <0 1>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_etr>;
-			};
-
-			trig-conns@1 {
-				reg = <1>;
-				arm,trig-in-sigs = <2 3>;
-				arm,trig-in-types = <SNK_FULL
-						     SNK_ACQCOMP>;
-				arm,trig-out-sigs = <2 3>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_etf>;
-			};
-
-			trig-conns@2 {
-				reg = <2>;
-				arm,trig-out-sigs = <4 5>;
-				arm,trig-out-types = <SNK_FLUSHIN
-						      SNK_TRIGIN>;
-				arm,cs-dev-assoc = <&cs_tpiu>;
-			};
-
-			trig-conns@3 {
-				reg = <3>;
-				arm,trig-in-sigs = <4 5 6 7>;
-				arm,trig-in-types = <STM_TOUT_SPTE
-						     STM_TOUT_SW
-						     STM_TOUT_HETE
-						     STM_ASYNCOUT>;
-				arm,cs-dev-assoc = <&cs_stm>;
-			};
-		};
+				trig-conns@0 {
+					reg = <0>;
+					arm,trig-in-sigs = <0 1>;
+					arm,trig-in-types = <SNK_FULL
+							     SNK_ACQCOMP>;
+					arm,trig-out-sigs = <0 1>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_etr>;
+				};
 
-		cs_cti1: cti@4a0a0000 {
-			compatible = "arm,coresight-cti", "arm,primecell";
-			reg = <0x4a0a0000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
+				trig-conns@1 {
+					reg = <1>;
+					arm,trig-in-sigs = <2 3>;
+					arm,trig-in-types = <SNK_FULL
+							     SNK_ACQCOMP>;
+					arm,trig-out-sigs = <2 3>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_etf>;
+				};
+
+				trig-conns@2 {
+					reg = <2>;
+					arm,trig-out-sigs = <4 5>;
+					arm,trig-out-types = <SNK_FLUSHIN
+							      SNK_TRIGIN>;
+					arm,cs-dev-assoc = <&cs_tpiu>;
+				};
 
-			trig-conns@0 {
-				reg = <0>;
-				arm,trig-in-sigs = <0>;
-				arm,trig-in-types = <GEN_IO>;
-				arm,trig-out-sigs = <0>;
-				arm,trig-out-types = <GEN_IO>;
-				arm,trig-conn-name = "dbtrgio";
-			};
-
-			trig-conns@1 {
-				reg = <1>;
-				arm,trig-out-sigs = <1 2>;
-				arm,trig-out-types = <STM_HWEVENT
-						      STM_HWEVENT>;
-				arm,cs-dev-assoc = <&cs_stm>;
+				trig-conns@3 {
+					reg = <3>;
+					arm,trig-in-sigs = <4 5 6 7>;
+					arm,trig-in-types = <STM_TOUT_SPTE
+							     STM_TOUT_SW
+							     STM_TOUT_HETE
+							     STM_ASYNCOUT>;
+					arm,cs-dev-assoc = <&cs_stm>;
+				};
 			};
-		};
 
-		cs_cpu_debug0: cpu-debug@4a210000 {
-			compatible = "arm,coresight-cpu-debug", "arm,primecell";
-			reg = <0x4a210000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			status = "disabled";
-		};
+			cs_cti1: cti@4a0a0000 {
+				compatible = "arm,coresight-cti", "arm,primecell";
+				reg = <0x4a0a0000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
 
-		cs_cti_cpu0: cti@4a220000 {
-			compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
-				     "arm,primecell";
-			reg = <0x4a220000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			cpu = <&cpu0>;
-			arm,cs-dev-assoc = <&cs_etm0>;
-			status = "disabled";
-		};
+				trig-conns@0 {
+					reg = <0>;
+					arm,trig-in-sigs = <0>;
+					arm,trig-in-types = <GEN_IO>;
+					arm,trig-out-sigs = <0>;
+					arm,trig-out-types = <GEN_IO>;
+					arm,trig-conn-name = "dbtrgio";
+				};
 
-		cs_etm0: etm@4a240000 {
-			compatible = "arm,coresight-etm4x", "arm,primecell";
-			reg = <0x4a240000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk", "atclk";
-			cpu = <&cpu0>;
-			status = "disabled";
+				trig-conns@1 {
+					reg = <1>;
+					arm,trig-out-sigs = <1 2>;
+					arm,trig-out-types = <STM_HWEVENT
+							      STM_HWEVENT>;
+					arm,cs-dev-assoc = <&cs_stm>;
+				};
+			};
+
+			cs_cpu_debug0: cpu-debug@4a210000 {
+				compatible = "arm,coresight-cpu-debug", "arm,primecell";
+				reg = <0x4a210000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+			};
+
+			cs_cti_cpu0: cti@4a220000 {
+				compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
+					     "arm,primecell";
+				reg = <0x4a220000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+				clock-names = "apb_pclk";
+				cpu = <&cpu0>;
+				arm,cs-dev-assoc = <&cs_etm0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+			};
 
-			out-ports {
-				port {
-					etm0_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port0>;
+			cs_etm0: etm@4a240000 {
+				compatible = "arm,coresight-etm4x", "arm,primecell";
+				reg = <0x4a240000 0x1000>;
+				clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
+				clock-names = "apb_pclk", "atclk";
+				cpu = <&cpu0>;
+				access-controllers = <&dbg_bus 0>;
+				status = "disabled";
+
+				out-ports {
+					port {
+						etm0_out_port: endpoint {
+							remote-endpoint = <&funnel_in_port0>;
+						};
 					};
 				};
 			};
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -37,45 +37,6 @@
 			     <GIC_PPI 11 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>,
 			     <GIC_PPI 10 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
 	};
-
-	soc@0 {
-		cs_cpu_debug1: cpu-debug@4a310000 {
-			compatible = "arm,coresight-cpu-debug", "arm,primecell";
-			reg = <0x4a310000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			cpu = <&cpu1>;
-			status = "disabled";
-		};
-
-		cs_cti_cpu1: cti@4a320000 {
-			compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti",
-				     "arm,primecell";
-			reg = <0x4a320000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>;
-			clock-names = "apb_pclk";
-			cpu = <&cpu1>;
-			arm,cs-dev-assoc = <&cs_etm1>;
-			status = "disabled";
-		};
-
-		cs_etm1: etm@4a340000 {
-			compatible = "arm,coresight-etm4x", "arm,primecell";
-			reg = <0x4a340000 0x1000>;
-			clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
-			clock-names = "apb_pclk", "atclk";
-			cpu = <&cpu1>;
-			status = "disabled";
-
-			out-ports {
-				port {
-					etm1_out_port: endpoint {
-						remote-endpoint = <&funnel_in_port1>;
-					};
-				};
-			};
-		};
-	};
 };
 
 &cs_funnel {
@@ -87,6 +48,47 @@
 			};
 		};
 	};
+};
+
+&dbg_bus {
+	cs_cpu_debug1: cpu-debug@4a310000 {
+		compatible = "arm,coresight-cpu-debug", "arm,primecell";
+		reg = <0x4a310000 0x1000>;
+		clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+		clock-names = "apb_pclk";
+		cpu = <&cpu1>;
+		access-controllers = <&dbg_bus 0>;
+		status = "disabled";
+	};
+
+	cs_cti_cpu1: cti@4a320000 {
+		compatible = "arm,coresight-cti-v8-arch", "arm,coresight-cti", "arm,primecell";
+		reg = <0x4a320000 0x1000>;
+		clocks = <&scmi_clk CK_SCMI_SYSDBG>;
+		clock-names = "apb_pclk";
+		cpu = <&cpu1>;
+		arm,cs-dev-assoc = <&cs_etm1>;
+		access-controllers = <&dbg_bus 0>;
+		status = "disabled";
+	};
+
+	cs_etm1: etm@4a340000 {
+		compatible = "arm,coresight-etm4x", "arm,primecell";
+		reg = <0x4a340000 0x1000>;
+		clocks = <&scmi_clk CK_SCMI_SYSDBG>, <&scmi_clk CK_SCMI_SYSATB>;
+		clock-names = "apb_pclk", "atclk";
+		cpu = <&cpu1>;
+		access-controllers = <&dbg_bus 0>;
+		status = "disabled";
+
+		out-ports {
+			port {
+				etm1_out_port: endpoint {
+					remote-endpoint = <&funnel_in_port1>;
+				};
+			};
+		};
+	};
 };
 
 &intc {
